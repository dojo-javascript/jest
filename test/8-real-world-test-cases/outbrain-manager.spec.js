const { create } = require('../../app/outbrain-manager')
    
jest.mock('../../app/script-detect', () => ({
  alreadyExist: jest.fn(() => true)
}))

describe('Outbrain manager', () => {
  
  let container

  beforeEach(() => {
    container = document.createElement('div')
  
    document.body.appendChild(container)
  })

  afterEach(() => {
    document.body.removeChild(container)
    jest.clearAllMocks()
  })

  it('should not init if no placeholder', () => {

    const { alreadyExist } = require('../../app/script-detect')

    create()

    expect(alreadyExist).not.toHaveBeenCalled()
  })

  describe('with placeholder', () => {

    beforeEach(() => {
      container.innerHTML = `
        <div data-outbrain-container></div>
      `
    })

    afterEach(() => {
      jest.clearAllMocks()
    })
    
    it('should inject outbrain script', () => {
  
      const { alreadyExist } = require('../../app/script-detect')
  
      expect(alreadyExist).not.toHaveBeenCalled()

      create()
  
      expect(alreadyExist).toHaveBeenCalled()

      expect(container).toMatchSnapshot()
    })
    
    it('should inject outbrain script', () => {
  
      container.innerHTML = `
        <div data-outbrain-container 
             data-ob-template="my-product"
             data-src="https://my-website.other.com"
             data-widget-id="SOME_ID"></div>
      `
      const { alreadyExist } = require('../../app/script-detect')

      create()
  
      expect(alreadyExist).toHaveBeenCalled()

      expect(container).toMatchSnapshot()
    })

  })

})