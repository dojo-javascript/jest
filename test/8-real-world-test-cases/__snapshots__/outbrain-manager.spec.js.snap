// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outbrain manager with placeholder should inject outbrain script 1`] = `
<div>
  
        
  <div
    class="OUTBRAIN"
    data-ob-template="my-project"
    data-src="https://my-website.com"
    data-widget-id="CRMB_1"
  />
  
      
</div>
`;

exports[`Outbrain manager with placeholder should inject outbrain script 2`] = `
<div>
  
        
  <div
    class="OUTBRAIN"
    data-ob-template="my-product"
    data-src="https://my-website.other.com"
    data-widget-id="SOME_ID"
  />
  
      
</div>
`;
